operator-framework-core/src/main/java/io/javaoperatorsdk/operator/api/reconciler/PrimaryUpdateAndCacheUtils.java

```diff
@@ -229,4 +236,152 @@ private static <P extends HasMetadata> P pollLocalCache(
       throw new OperatorException(e);
     }
   }
+
+  /** Adds finalizer using JSON Patch. Retries conflicts and unprocessable content (HTTP 422) */
+  @SuppressWarnings("unchecked")
+  public static <P extends HasMetadata> P addFinalizer(Context<P> context, String finalizer) {
```

Comment:
Please add complete documentation, in particular with respect to the return value since it shouldn't be left up to users what is returned.